src/main/java/org/spongepowered/api/text/conversation/ExternalChatHandler.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.text.conversation;
+
+import org.spongepowered.api.text.Text;
+
+/**
+ * Each {@link Conversant} has one of these to handle the chat messages being
+ * sent to them.
+ */
+public interface ExternalChatHandler {
```

Comment:
Are you suggesting that he should add a ECHType that is a factory like think that can create ECH instances? Because ECH itself cannot be CatalogTypes because they exists as 1/conversant and not 1/game. The id needs to be unique and must contain a meaning, just iterating some counters and throwaway names is not the purpose of CatalogTypes.

However a factory like thing is required anyway because the ECH is not target aware but is returned in the archetpye for all conversants.